src/lib.rs

```diff
@@ -21,16 +20,19 @@ impl Default for EthKeys {
     }
 }
 
-// Implement the contract structure
 #[near_bindgen]
 impl EthKeys {
     // Sets the encrypted key for the sender's account.
-    pub fn set_key(&mut self, encrypted_key: String) {
+    pub fn set_key(&mut self, encrypted_key: String, overwrite: bool) {
         // TODO - would be nice if there was some way to validate
         // that the encrypted key actualy contains expected data.
         let account_id = env::signer_account_id();
```

Comment:
There are nuances around `signer_account_id` and `predecessor_account_id`, I would have to verify, but this way a rogue smart contract might be able to set a new key for a user. It's also common practice to require a yoctoNEAR deposits, which can only be done using FullAccessKeys, but not with FunctionCallKeys.

Reply:
Very interesting. Thanks. I was under the impression that the yoco deposit would be a separate change that would also have to make the function payable. I had captured this in issue https://github.com/Mintbase/neareth/issues/2. However, your comment seems to mean two different things here (something about how other accounts can overwrite other's keys)... Are these two different things, or does having attached deposit somehow prevent this?

Reply:
Fixed in 726d486

Reply:
#2 covers everything, except the `signer`/`predecessor`. Most of this comment is just information about NEAR smart contracts and common practices.

Reply:
So then would you say it's "approved"?

Reply:
Left a comment on #2, I think it's better addressed there.